engine/src/main/java/org/destinationsol/systems/DamageSystem.java

```diff
@@ -50,6 +55,10 @@ public EventResult onDamage(DamageEvent event, EntityRef entity) {
             }
             health.currentHealth = newHealthAmount;
             entity.setComponent(health);
+
+            if (health.currentHealth <= 0) {
```

Comment:
This whole block can be simplified, no? You're effectively checking for negative health twice now.

Reply:
See my earlier reply to Adrijaned. I think that there are two different things that happen when an entity with 30 health takes 40 damage. 
1) The entity's health drops to 0 (so it shouldn't display -10 for the hero, for example)
2) a DestroyEvent is created
I think that the two things aren't intrinsically linked. If an entity doesn't get removed by the DestroyEvent, its health still should not be negative. Does that make sense? If so, I'll add some Javadoc explaining that.

Reply:
Nope, this block can't be simplified, since you need to set the Health component on the entity before sending the event

Reply:
Ok that makes sense, comments explaining that would be great :)